src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -133,7 +133,7 @@ class FilterButton extends React.Component<Props, State> {
                 [fieldDisplayTextType]: fieldDisplayText,
                 [fieldKeyType]: fieldKey,
                 valueType,
-                fieldId,
+                fieldId: id,
```

Comment:
expected?

Reply:
Yes, ConditionType has a property called fieldId but OptionType has a property called id. The right hand of the assignment is coming from OptionType.

Reply:
What does Condition.fieldId = Option.id mean? It isn't obvious to me.